internal/testutils/podbuilder/podbuilder.go

```diff
@@ -0,0 +1,207 @@
+package podbuilder
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/docker/distribution/reference"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/windmilleng/tilt/internal/container"
+	"github.com/windmilleng/tilt/internal/k8s"
+	"github.com/windmilleng/tilt/internal/model"
+)
+
+const FakeDeployID = model.DeployID(1234567890)
+const FakeContainerID = "myTestContainer"
+
+// Builds Pod objects for testing
+//
+// Ensures that any PodStatus objects we create are consistent with the Pod in
+// the PodTemplateSpec in the manifest.
+type PodBuilder struct {
+	t        testing.TB
+	manifest model.Manifest
+
+	podID        string
+	phase        string
+	creationTime time.Time
+	deployID     model.DeployID
+
+	// keyed by container index
+	imageRefs map[int]string
+	cIDs      map[int]string
+}
+
+func New(t testing.TB, manifest model.Manifest) PodBuilder {
+	return PodBuilder{
+		t:         t,
+		manifest:  manifest,
+		imageRefs: make(map[int]string),
+		cIDs:      make(map[int]string),
+	}
+}
+
+func (b PodBuilder) WithPodID(podID string) PodBuilder {
+	msgs := validation.NameIsDNSSubdomain(podID, false)
+	if len(msgs) != 0 {
+		b.t.Fatalf("pod id %q is invalid: %s", podID, msgs)
+	}
+	b.podID = podID
+	return b
+}
+
+func (b PodBuilder) WithPhase(phase string) PodBuilder {
+	b.phase = phase
+	return b
+}
+
+func (b PodBuilder) WithImage(image string, index int) PodBuilder {
```

Comment:
maybe call this `WithImageAtIndex` and a separate func to set the image (/container id) for Container[0], since imagine that'll be the majority of invocations of this function, and the extra parameter for index is kinda confusing